sqlalchemy_filters/models.py

```diff
@@ -30,6 +31,17 @@ def get_query_models(query):
         A dictionary with all the models included in the query.
     """
     models = [col_desc['entity'] for col_desc in query.column_descriptions]
+    if (
+        hasattr(query, '_select_from_entity') and
+        (query._select_from_entity is not None) and
+        (query._select_from_entity not in models)
+    ):
+        if isinstance(query._select_from_entity, Mapper):  # pragma: no cover
+            # sqlalchemy>=1.1
+            models.append(query._select_from_entity.class_)
+        else:  # pragma: no cover
+            # sqlalchemy==1.0
+            models.append(query._select_from_entity)
```

Comment:
Could the `query._select_from_entity` model be already present in the `models` list?

`models` has been generated from `query.column_descriptions` and we could potentially check that `type(query._select_from_entity)` is not the type of any of the objects already present in `models`.

I am just trying to avoid adding a duplicate model to the `models` list, in case that's possible.

Reply:
sure. i'm really sorry to waste you time, whole block was indeed wrong because the presence check was done only by sqla 1.1+ API.